validmind/vm_models/dataset.py

```diff
@@ -407,7 +411,7 @@ def y_pred(self, model_id) -> np.ndarray:
                 for name in self.columns
                 if name == self.prediction_column(model_id=model_id)
             ],
-        ]
+        ].flatten()
```

Comment:
This was key. We need to ensure we don't end up with numpy arrays inside numpy arrays.

Reply:
thanks for this.